Source/WebCore/dom/Observable.cpp

```diff
@@ -70,16 +71,13 @@ void Observable::subscribe(ScriptExecutionContext& context, std::optional<Observ
         subscribeInternal(context, InternalObserverFromScript::create(context, nullptr), options);
 }
 
-void Observable::subscribeInternal(ScriptExecutionContext& context, Ref<InternalObserver> observer, SubscribeOptions options)
+void Observable::subscribeInternal(ScriptExecutionContext& context, Ref<InternalObserver> observer, const SubscribeOptions& options)
 {
     RefPtr document = dynamicDowncast<Document>(context);
     if (document && !document->isFullyActive())
         return;
 
-    auto subscriber = Subscriber::create(context, observer);
-
-    if (options.signal)
-        subscriber->followSignal(*options.signal.get());
+    auto subscriber = Subscriber::create(context, observer, options);
```

Comment:
The Subscriber should own/hold a reference to the options. `SubscribeOptions` is a simple struct, and does not implement `ActiveDOMObject` as such, we will pass by reference, and store the value inside.

cc @cdumez

Reply:
`auto` -> `Ref`